pkg/objstore/s3/s3.go

```diff
@@ -192,6 +207,46 @@ func (b *Bucket) Name() string {
 	return b.name
 }
 
+// startCredentialExpirer will start a watch of the file directory of AWS credentials and
+// if it is updated it will force expire credentials so they are fetched again next get.
+func startCredentialExpirer(creds *credentials.Credentials, filePath string) error {
+	// Will not start watch for empty filePath
+	if filePath == "" {
+		return nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+
+	// Launch the watch which will expire the credentials when fired
+	go func() {
```

Comment:
We need to kill this go routine when thanos quit or dies. Is there a standard way thanos does this atm for other go routines?